src/swoop/api/io.py

```diff
@@ -0,0 +1,88 @@
+import io
+import logging
+
+from minio import Minio
+from minio.error import S3Error
+
+logger = logging.getLogger("uvicorn")
+logger.setLevel(logging.INFO)
+
+
+class IOClient:
+    client = None
+    bucket_name = None
+
+    def __init__(
+        self, s3_endpoint: str, access_key: str, secret_key: str, bucket_name=None
+    ):
+        """Initialize IO Client."""
+        self.client = Minio(
+            s3_endpoint.split("//").pop() if "http" in s3_endpoint else s3_endpoint,
+            access_key=access_key,
+            secret_key=secret_key,
+            secure="http://" not in s3_endpoint,
+        )
+        self.bucket_name = bucket_name
+
+    def set_bucket_name(self, bucket_name: str):
+        self.bucket_name = bucket_name
+
+    def get_object(self, object_name: str):
+        """Retrieve from object storage."""
+        object_response = None
+        if (
+            object_name
+            and self.bucket_name
+            and self.client.bucket_exists(self.bucket_name)
+        ):
+            try:
+                try:
+                    response = self.client.get_object(self.bucket_name, object_name)
+                    object_response = response.json()
+                    logger.debug(f"retrieved object content: {object_response}")
```

Comment:
Very slight nitpick, take it or leave it: I'd minimize the amount of code you have in the try. Of course I don't think either of these lines would raise the handled exception, but someone might make code changes in the future and follow this pattern.

It seems like you don't want these lines to run if the exception is triggered, so you could use an `else` with the `try`, to allow you to move these lines out of the `try`.